sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -547,14 +478,16 @@ object IntervalUtils {
             case ' ' =>
               fraction /= NANOS_PER_MICROS.toInt
               state = TRIM_BEFORE_UNIT
-            case _ => return null
+            case _ if '0' <= b && b <= '9' =>
+              throwIAE(s"invalid value fractional part '$fraction$nextWord' out of range")
```

Comment:
how about `interval can only support nanosecond precision, '$nextWord' is out of range`

Reply:
BTW can we really implement `nextWord` correctly? We need to know where we start to parse a number, seems we don't track it now.

Reply:
> how about `interval can only support nanosecond precision, '$nextWord' is out of range`
the is not suitable, `0.9999999999` the nextword will be `9` only but '$fraction$nextWord' is the exact `9999999999`